### PR TITLE
initial commit refactor task

### PR DESCRIPTION
**Pull request by cedrikvonheiroth**

### CODE REVIEW

File: algolia-index-js-searchpage.php [hunk 1]
```diff
@@ -14,13 +14,21 @@
...
 }
 
 define('ALGOLIAINDEXJSSEARCHPAGE_PATH', plugin_dir_path(__FILE__));
 define('ALGOLIAINDEXJSSEARCHPAGE_URL', plugins_url('', __FILE__));
 define('ALGOLIAINDEXJSSEARCHPAGE_TEMPLATE_PATH', ALGOLIAINDEXJSSEARCHPAGE_PATH . 'templates/');
+define('ALGOLIA_INDEX_MOUNT_POINT', 'custom_search_page');
+
+
+define('ALGOLIAINDEXJSSEARCHPAGE_VIEW_PATH', ALGOLIAINDEXJSSEARCHPAGE_PATH . 'views/');
+define('ALGOLIAINDEXJSSEARCHPAGE_MODULE_VIEW_PATH', plugin_dir_path(__FILE__) . 'source/php/Modules');
```
Comment (sebastianthulin) [inline algolia-index-js-searchpage.php:29]: No modules in this plugin. This constant will not be used.

File: algolia-index-js-searchpage.php [hunk 2]
```diff
@@ -14,13 +14,21 @@
...
+define('ALGOLIAINDEXJSSEARCHPAGE_MODULE_PATH', ALGOLIAINDEXJSSEARCHPAGE_PATH . 'source/php/Modules/');
```
Comment (sebastianthulin) [inline algolia-index-js-searchpage.php:30]: No modules in this plugin. This constant will not be used.

File: package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
     "@babel/preset-react": "^7.12.10",
     "algoliasearch": "^4.12.1",
     "he": "^1.2.0",
+    "instantsearch.js": "^4.41.2",
     "qs": "^6.9.6",
     "react": "^16.14.0",
     "react-instantsearch-dom": "^6.8.3"
```
Comment (sebastianthulin) [inline package.json:24]: Remove this package

File: source/php/App.php [hunk 4]
```diff
@@ -81,15 +81,28 @@ public function enqueueScripts()
...
+    {
+        if (!self::isSearchPage()) {
+            return;
         }
 
-        global $renderedSearch; 
-        if(!isset($renderedSearch)) {
-            $renderedSearch = true; 
-            echo '<div id="algolia-instantsearch-react"><div class="c-algolia-instantsearch-ripple" data-label="' . __("Searching", 'algolia-index-js-searchpage') . '"><div></div><div></div></div></div>'; 
+        global $renderedSearch;
```
Comment (sebastianthulin) [inline source/php/App.php:90]: May be moved to a blade rendered file for clarity.

File: source/php/App.php [hunk 5]
```diff
@@ -98,18 +111,19 @@ public function renderSearchpageMount($query) {
...
+    {
 
         if (is_multisite() && (defined('SUBDOMAIN_INSTALL') && SUBDOMAIN_INSTALL === false)) {
             if (trim(strtok($_SERVER["REQUEST_URI"], '?'), "/") == trim(get_blog_details()->path, "/") && is_search()) {
                 return true;
             }
         }
 
-        if (trim(strtok($_SERVER["REQUEST_URI"], '?'), "/") == "" && is_search()) {
+        if (trim(strtok($_SERVER["REQUEST_URI"], '?'), "/") == "wptest" && is_search()) {
```
Comment (sebastianthulin) [inline source/php/App.php:123]: Debug data?

File: source/php/Helper/React.php [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?php
```
Comment (sebastianthulin) [inline source/php/Helper/React.php:2]: Remove file

File: source/php/App.php (hunk 5 above)
Comment (NiclasNorin) [reply source/php/App.php:123]: We have a small difference in our database working locally. We will make sure this one is empty when launching

**Pull request merged**